### ISSUE TITLE
Display dates in local time

### ISSUE THREAD

**Issue by cco3** [aboutcode-org/scancode.io#142]
The Web UI currently displays thing like the "Created time", etc. using UTC time, but it might make sense to use local time or both UTC and local time.

(I was having trouble figuring out which scan I was looking at..."was this the one I started at 8:00 or at 11:00?")

**Comment by tdruez** [comment]
@cco3 Summary of changes about "dates" in latest version:

- The local installation timezone can now be configured using the TIME_ZONE settings. See https://scancodeio.readthedocs.io/en/latest/scancodeio-settings.html#time-zone for details.
- The current timezone is now displayed along the date in the web UI.
- The creation date of a project is displayed as a more readable "Created 6 hours ago", the full date information can be displayed hovering the date label:
![date](https://user-images.githubusercontent.com/489057/116404736-3a4d3600-a82f-11eb-9df3-84dbaf9cd646.jpg)

**Comment by cco3** [comment]
By "The current timezone", do you mean the timezone specified in the settings?  Why not just let the client display the time according to the local timezone?

**Comment by tdruez** [comment]
@cco3 

> By "The current timezone", do you mean the timezone specified in the settings?

Yes, Django allows to specify the timezone of the server, not to get the timezone for each clients. The dates are always stored as UTC in the database and displayed using the server defined timezone (`TIME_ZONE` settings).

> Why not just let the client display the time according to the local timezone?

If by "local timezone" you mean "client timezone" rather than "server timezone", how would you get that information from the client?

**Comment by cco3** [comment]
The server serves up UTC.  The client then displays it according to the local timezone.  new Date(utcTimestamp).toString() would be one example that works in chrome.

Unfortunately the standard javascript time functions are very inconsistent across browsers, but there are libraries commonly used to fix that.

**Comment by cco3** [comment]
Times are still being shown in UTC:

![Screenshot from 2021-08-06 11-32-21](https://user-images.githubusercontent.com/236057/128556321-a3b04280-7e9a-416e-81ca-83ba49c7ac19.png)

**Comment by lf32** [comment]
It works on setting the value for the `TIME_ZONE` variable in the env

![time_env_zone](https://user-images.githubusercontent.com/96695352/163559466-50e8acb0-1170-42db-a1fd-b8c564a95116.png)

when variable value is set to IST 

![time_ist](https://user-images.githubusercontent.com/96695352/163559777-86246fb2-a975-4443-bfed-4607288f9de9.png)

when variable has no value set it's UTC

![time_utc](https://user-images.githubusercontent.com/96695352/163559829-ce995525-ce67-4032-828d-284d3b8cb186.png)

**Comment by lf32** [comment]
why geo when value can be set directly in the settings?

**Comment by pombredanne** [comment]
@lf32 Thank you for the excellent research!
Note that's changing the date system-wide including in the DB... which may not always be what one wants and may be keeping UTC there is simpler... The alternative is only to display a "local time" based on a web user date 
The trick is to get the user date/time setting which is not something Django handles and is typically best done clien-side with JS afaik. See for instance https://github.com/adamcharnock/django-tz-detect

See  also https://docs.djangoproject.com/en/4.0/topics/i18n/timezones/#template-tags

@cco3 Would be changing  the setting be enough for you?

**Comment by lf32** [comment]
@pombredanne Thank you, I'll check them
